OWML.ModHelper.Menus/ModInputCombinationElementMenu.cs

```diff
@@ -0,0 +1,202 @@
+            GameObject.Destroy(originalMenu);
+            GameObject.Destroy(_inputMenu.GetValue<Text>("_labelText").GetComponent<LocalizedText>());
+            Initialize((Menu)_inputMenu);
+        }
+
+        public void Open(string value, string comboName, IModInputCombinationMenu combinationMenu = null, IModInputCombinationElement element = null)
+        {
+            _combinationMenu = combinationMenu;
+            _element = element;
+            _comboName = comboName;
+            _inputMenu.OnPopupConfirm += OnPopupConfirm;
+            _inputMenu.OnPopupCancel += OnPopupCancel;
+            _inputMenu.OnPopupValidate += OnPopupValidate;
+
+            const string message = "Press your combination";
+
+            _inputMenu.EnableMenu(true, value);
+
+            var okCommand = InputLibrary.confirm;
+            var okPrompt = new ScreenPrompt(okCommand, "OK");
+            if (_cancelCommand == null)
+            {
+                _cancelCommand = new SingleAxisCommand();
+                var cancelBindingGamepad = new InputBinding(JoystickButton.Select);
+                var cancelBindingKeyboard = new InputBinding(KeyCode.Escape);
+                _cancelCommand.SetInputs(cancelBindingGamepad, cancelBindingKeyboard);
+                var commandObject = new GameObject();
+                var commandComponent = commandObject.AddComponent<ModCommandUpdater>();
+                commandComponent.Initialize(_cancelCommand);
+            }
+            var cancelPrompt = new ScreenPrompt(_cancelCommand, "Cancel");
+            var resetPrompt = new ScreenPrompt("Reset");
+            _inputMenu.SetUpPopup(message, okCommand, _cancelCommand, null, okPrompt, cancelPrompt, resetPrompt);
+            _inputMenu.GetValue<Text>("_labelText").text = message;
+        }
+
+        private void RerouteToConsole(string message)
+        {
+            Console.WriteLine("Failed to create popup for a following message:");
+            Console.WriteLine(message);
+        }
+
+        private bool OnPopupValidate()
+        {
+            var currentCombination = _inputMenu.Combination;
+            var collisions = _inputHandler.GetCollisions(currentCombination);
+            if (collisions.Count > 0 && collisions[0] != _comboName)
+            {
+                if (_twoButtonPopup == null)
+                {
+                    RerouteToConsole($"This combination collides with \"{collisions[0]}\"");
+                    return false;
+                }
+                _twoButtonPopup.EnableMenu(true);
+                _twoButtonPopup.SetUpPopup($"This combination collides with \"{collisions[0]}\"", InputLibrary.confirm2, null,
+                    new ScreenPrompt(InputLibrary.confirm2, "Ok"), new ScreenPrompt("Cancel"), true, false);
+                _twoButtonPopup.GetValue<Text>("_labelText").text = $"this combination collides with \"{collisions[0]}\"";
```

Comment:
Same message three times, extract to variable

Reply:
K